web_responsive/static/tests/js/web_responsive.js

```diff
@@ -1,278 +1,317 @@
 /* Copyright 2016 LasLabs Inc.
  * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
 
-odoo.define_section('web_responsive', ['web_responsive'], function(test) {
+odoo.define('web_responsive.test', function(require) {
     "use strict";
-    
-    // It provides a base drawer compatible interface for testing
-    self.initInterface = function(AppDrawer) {
-        
-        var $el = $('<div class="drawer drawer--left">');
-        $el.append(
-            $('<header role="banner">')
-                .append(
-                    $('<button class="drawer-toggle"><span class="drawer-hamburger-icon">')
-                )
-                .append(
-                    $('<nav class="drawer-nav"><ul class="drawer-menu"><li class="drawer-menu-item">')
-                )
-                .append(
-                    $('<div class="panel-title" id="appDrawerAppPanelHead">')
-                )
-        ).append($('<main role="main">'));
-        
-        self.$clickZone = $('<a class="oe_menu_leaf">');
-        
-        self.$secondaryMenu = $('<div><div class="o_sub_menu_content">');
-        
-        self.$dropdown = $('<div class="dropdown-scrollable">');
-    
-        $el.append(self.$clickZone)
-            .append(self.$secondaryMenu)
-            .append(self.$dropdown);
-        
-        var $document = $("#qunit-fixture");
-        $document.append($el);
-        
-        self.drawer = new AppDrawer.AppDrawer();
-        
-        return $document;
-        
-    };
-    
-    self.linkGrid = function() {
-        for(var i=0; i < 3; i++){
-            self.drawer.$el.append(
-                $('<div class="row">').append(
-                    $('<a class="col-md-6" id="a_' + i + '"><span class="app-drawer-icon-app /></a>' +
-                      '<a class="col-md-6" id="b_' + i + '"><span class="app-drawer-icon-app /></a>'
-                      )
-                )
-            );
-            self.drawer.$appLinks = $('a.col-md-6');
+
+    var core = require('web.core');
+    var responsive = require('web_responsive');
+
+    QUnit.module('web_responsive', {
+        beforeEach: function() {
+            var $el = $('<div class="drawer drawer--left">');
+
+            $el.append(
+                $('<header role="banner">')
+                    .append(
+                        $('<button class="drawer-toggle"><span class="drawer-hamburger-icon">')
+                    )
+                    .append(
+                        $('<nav class="drawer-nav"><ul class="drawer-menu"><li class="drawer-menu-item">')
+                    )
+                    .append(
+                        $('<div class="panel-title" id="appDrawerAppPanelHead">')
+                    )
+            ).append($('<main role="main">'));
```

Comment:
I think it's better to just bundle all of this appends in the single first `var $el = ...` definition. KISS.